bfv/encryptor.go

```diff
@@ -293,5 +304,38 @@ func (encryptor *skEncryptor) encrypt(plaintext *Plaintext, ciphertext *Cipherte
 	encryptor.gaussianSamplerQ.ReadAndAdd(ciphertext.value[0])
 
 	// ct = [-a*s + m + e , a]
-	ringQ.Add(ciphertext.value[0], plaintext.value, ciphertext.value[0])
+	if !plaintext.inZQ {
+		encryptor.TtoQ(plaintext)
+		encryptor.ringQ.Add(ciphertext.value[0], encryptor.polypool[0], ciphertext.value[0])
+	} else {
+		encryptor.ringQ.Add(ciphertext.value[0], plaintext.value, ciphertext.value[0])
+	}
+}
+
+func (encryptor *encryptor) TtoQ(p *Plaintext) {
+
+	ringQ := encryptor.ringQ
+	encryptor.polypool[0].Zero()
+	for i := len(ringQ.Modulus) - 1; i >= 0; i-- {
```

Comment:
Shouldn't this operation be part of the `ring` package ?

Reply:
To be seen at a later stage if we want to further generalize the operations to vectors, but this would require some work directly on the ring package. For now I'd leave like this.